### PR TITLE
Re-add fh streaming possibility

### LINKED ISSUE #36: Streaming decryption
Any plans to support streaming decryption from a file-like object? We want to stream large encrypted files from an sftp server and decrypt them on the fly as we write them somewhere else.

It looks like `decrypt_file` is basically doing that, we just need an API that takes the actual file-like fileobj.
https://github.com/woodruffw/pyrage/blob/aab0f86bca80eb263b8d7b5fa4fdb06dfb52f095/src/lib.rs#L224

Maybe using something like this https://github.com/omerbenamram/pyo3-file

### PR DESCRIPTION
**Pull request by julianhille**
Adding io input back in.

No idea if that is of any help for anyone else, but at least i need to be able to stream a pyfilelike object.

Feel free to roast all of it as this is my first contact with rust and rust as python module. :>

Fixes #36

### CODE REVIEW

**Comment by woodruffw** [comment]
Thanks @julianhille, I'll try and give this a review today or tomorrow.

**Comment by woodruffw** [comment]
Looks like the lint is failing; mind running `cargo fmt` locally?

File: Cargo.toml [hunk 1]
```diff
@@ -25,3 +25,4 @@ pyo3 = { version = "0.22.3", features = [
     "abi3-py38",
     "py-clone",
 ] }
+pyo3-file = "0.9.0"
```
Comment (woodruffw) [inline Cargo.toml:28]: Trailing newline here, please 🙂 

```suggestion
pyo3-file = "0.9.0"

```

File: pyrage-stubs/pyrage-stubs/__init__.pyi [hunk 2]
```diff
@@ -15,5 +16,7 @@ class IdentityError(Exception):
 
 def encrypt(plaintext: bytes, recipients: Sequence[Recipient]) -> bytes: ...
 def encrypt_file(infile: str, outfile: str, recipients: Sequence[Recipient]) -> None: ...
+def encrypt_io(in_fh: BufferedIOBase, recipients: Sequence[Recipient]) -> bytes: ...
 def decrypt(ciphertext: bytes, identities: Sequence[Identity]) -> bytes: ...
 def decrypt_file(infile: str, outfile: str, identities: Sequence[Identity]) -> None: ...
+def decrypt_io(in_fh: BufferedIOBase, outfile: BufferedIOBase, recipient: Sequence[Recipient]) -> None: ...
```
Comment (woodruffw) [inline pyrage-stubs/pyrage-stubs/__init__.pyi:22]:
```suggestion
def decrypt_io(in_fh: BufferedIOBase, outfile: BufferedIOBase, recipient: Sequence[Recipient]) -> None: ...

```

**Review by woodruffw — COMMENTED** [review]
Looks good overall, a few small style nits (plus the lint failures in CI)!

**Comment by julianhille** [comment]
> Looks like the lint is failing; mind running `cargo fmt` locally?

Just learned about rustfmt and Cargo fmt yesterday. Committed your suggestions, will rebase / Squash format and then get back to you. 

If thats OK with you to rebase in the middle of a review.

**Comment by julianhille** [comment]
Fixed and rebased. Let me know if there is anything else I can do.

**Comment by woodruffw** [comment]
Some more lint failures:

```
error: unused variable: `py`
   --> src/lib.rs:263:18
    |
263 | fn from_pyobject(py: Python, file: PyObject, read_only: bool) -> PyResult<PyFileLikeObject> {
    |                  ^^ help: if this is intentional, prefix it with an underscore: `_py`
    |
    = note: `-D unused-variables` implied by `-D warnings`
    = help: to override `-D warnings` add `#[allow(unused_variables)]`

error: unneeded `return` statement
   --> src/lib.rs:265:5
    |
265 |     return PyFileLikeObject::with_requirements(file, read_only, !read_only, false, false);
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_return
    = note: `-D clippy::needless-return` implied by `-D warnings`
    = help: to override `-D warnings` add `#[allow(clippy::needless_return)]`
help: remove `return`
    |
265 -     return PyFileLikeObject::with_requirements(file, read_only, !read_only, false, false);
265 +     PyFileLikeObject::with_requirements(file, read_only, !read_only, false, false)
    |

error: could not compile `pyrage` (lib) due to 2 previous errors
Error: Process completed with exit code 101.
```

`cargo clippy --fix` should fix those.

Also, don't worry about rebasing/squashing -- it's fine to just push, and I'll handle the squash while merging 🙂

**Comment by julianhille** [comment]
Ah thanks overlooked that. Will get to it.

**Comment by julianhille** [comment]
tried again. should work now :> thanks for your patience

**Comment by julianhille** [comment]
Would you mind start the ci run again? Thanks

**Review by woodruffw — APPROVED** [review]
Thanks @julianhille, looks good to me!

**Comment by woodruffw** [comment]
Let me know if you'd like a release for this. Otherwise, I'll likely do it the next time another feature PR comes in.

**Comment by julianhille** [comment]
I'd Love a Release. If Not to much effort.

**Comment by woodruffw** [comment]
Released with v1.2.0. Thanks for your hard work here!

**Pull request merged**